llvm/lib/Support/RISCVISAInfo.cpp

```diff
@@ -854,6 +895,30 @@ RISCVISAInfo::parseArchString(StringRef Arch, bool EnableExperimentalExtension,
                              "string must be lowercase");
   }
 
+  bool IsProfile = Arch.starts_with("rvi") || Arch.starts_with("rva") ||
+                   Arch.starts_with("rvb") || Arch.starts_with("rvm");
+  std::string NewArch;
```

Comment:
`NewArch` might be assigned twice, I'm wondering whether we can pre-reserve a chunk of memory beforehand to prevent 1 more memory allocation?